src/python/grpcio_tests/tests_aio/unit/connectivity_test.py

```diff
@@ -38,59 +46,65 @@ class TestConnectivityState(AioTestBase):
+                _block_until_certain_state(
+                    channel, grpc.ChannelConnectivity.TRANSIENT_FAILURE),
+                test_constants.SHORT_TIMEOUT)
 
     async def test_normal_backend(self):
-        channel = aio.insecure_channel(self._server_address)
-
-        current_state = channel.check_connectivity_state(True)
-        self.assertEqual(grpc.ChannelConnectivity.IDLE, current_state)
-
-        deadline = time.time() + test_constants.SHORT_TIMEOUT
+        async with aio.insecure_channel(self._server_address) as channel:
+            current_state = channel.get_state(True)
+            self.assertEqual(grpc.ChannelConnectivity.IDLE, current_state)
 
-        while current_state != grpc.ChannelConnectivity.READY:
-            current_state = await channel.watch_connectivity_state(
-                current_state, deadline - time.time())
-            self.assertIsNotNone(current_state)
-
-        await channel.close()
+            # Should not time out
+            await asyncio.wait_for(
+                _block_until_certain_state(channel,
+                                           grpc.ChannelConnectivity.READY),
+                test_constants.SHORT_TIMEOUT)
 
     async def test_timeout(self):
-        channel = aio.insecure_channel(self._server_address)
-
-        self.assertEqual(grpc.ChannelConnectivity.IDLE,
-                         channel.check_connectivity_state(False))
+        async with aio.insecure_channel(self._server_address) as channel:
+            self.assertEqual(grpc.ChannelConnectivity.IDLE,
+                             channel.get_state(False))
 
-        # If timed out, the function should return None.
-        self.assertIsNone(await channel.watch_connectivity_state(
-            grpc.ChannelConnectivity.IDLE, test_constants.SHORT_TIMEOUT))
-
-        await channel.close()
+            # If timed out, the function should return None.
+            with self.assertRaises(asyncio.TimeoutError):
+                await asyncio.wait_for(
+                    _block_until_certain_state(channel,
+                                               grpc.ChannelConnectivity.READY),
+                    test_constants.SHORT_TIMEOUT)
 
     async def test_shutdown(self):
         channel = aio.insecure_channel(self._server_address)
 
         self.assertEqual(grpc.ChannelConnectivity.IDLE,
-                         channel.check_connectivity_state(False))
+                         channel.get_state(False))
 
         await channel.close()
 
         self.assertEqual(grpc.ChannelConnectivity.SHUTDOWN,
-                         channel.check_connectivity_state(False))
+                         channel.get_state(True))
+
+        self.assertEqual(grpc.ChannelConnectivity.SHUTDOWN,
+                         channel.get_state(False))
+
+        # It can raise Exception since it is an usage error, but it should not
+        # segfault or abort.
+        with self.assertRaises(Exception):
```

Comment:
Can we at least be more narrow than just `Exception`?

Reply:
Narrowed it to `RuntimeError`.

We might change the type of exceptions soon. So, I didn't want to commit to any type of exceptions just yet.